src/Elastic.Apm/Report/Serialization/BodyStringSanitizerConverter.cs

```diff
@@ -22,18 +22,23 @@ public override void WriteJson(JsonWriter writer, object value, JsonSerializer s
 		{
 			if (value is string strValue)
 			{
-				var formValues = strValue.Split('&');
-
 				if (!FormatCheck(strValue))
 					writer.WriteValue(strValue);
 				else
 				{
+					var formValues = strValue.Split('&');
+
 					var sb = new StringBuilder();
 					foreach (var formValue in formValues)
 					{
 						var formsValueSplit = formValue.Split('=');
-						if (formsValueSplit.Length != 2)
-							continue;
+						if (formsValueSplit.Length != 2 || string.IsNullOrWhiteSpace(formsValueSplit[0])
```

Comment:
According to [String.Split docs](https://docs.microsoft.com/en-us/dotnet/api/system.string.split?redirectedfrom=MSDN&view=netframework-4.8#System_String_Split_System_Char___System_Int32_System_StringSplitOptions_) returned array should not contain `nulls` - it can be empty or contain empty strings but it cannot contain `null`s.

Reply:
Appreciate the info, but I'm not sure what you suggest here or what action I should take.

You suggest checking on formsValueSplit[0] just for the length? Honestly I feel it just makes the code look wired - would that make any significant difference?